solidity-mode.el

```diff
@@ -474,8 +474,9 @@ Highlight the 1st result."
 ;; solidity syntax table
 (defvar solidity-mode-syntax-table
   (let ((st (make-syntax-table)))
-    ;; '_' underscore is a valid part of a word
+    ;; Underscore ('_') and dollar sign ('$') are valid parts of a word.
     (modify-syntax-entry ?_ "w" st)
+    (modify-syntax-entry ?$ "w" st)
```

Comment:
Is there a way for me to test this PR? Do you have an example solidity snippet that will be different after this change?

Reply:
you can test with this example:

```solidity
contract Test {
  uint a_1 = 2;
  uint a$2 = 3;
}
```

The digit 2 in `a$2` is currently of a numeric color.

After applying my patch, its color should be similar to `a_1`.